doc/source/development/contributing.rst

```diff
@@ -24,6 +24,27 @@ and `good first issue
 where you could start out. Once you've found an interesting issue, you can
 return here to get your development environment setup.
 
+When you start working on an issue, it's a good idea to assign the issue to yourself,
```

Comment:
I'm a little hesitant to word it this way, maybe to even allow this. For sure sometimes there is work duplication, but on the flip side I feel like more often someone says they want to work on an issue but never do (or at least never put a PR up) which may be discouraging for people with a legitimate interesting in seeing things all the way through.

Reply:
I see your point. But not sure if we can do any better. The current workflow has the same problems you mention I think. And the additional one that it's very time consuming to find the issues that nobody claimed.

With this new workflow, should be very fast to find issues nobody claimed. And it also moves in a good direction to automatically unassign issues (which we could currently do by adding an extra comment saying that the issue is available, but that doesn't look like a great thing).

Would you prefer to just say "work on any issue, and don't tell anyone, the probability of someone else working on the same is very small"? Feels a bit weird, may be not nice if duplication increases, but we can try.

Reply:
No I just think the discouragement of claiming an issue but then never really advancing it would be a worse issue than duplication. I might be missing some PRs but I've only seen duplication on a handful of small PRs

Reply:
claiming issues for a sprint is probably ok

and this does allow searching for things that are unclaimed

but will have to be rigorous about in assigning things after a little

Reply:
My view is that now people is already locking issues for themselves using a comment. And I assume most people don't start working on something if they see someone else said they are working on it.

For me that's reasonable. The problem that I see is that there is no way to get the list of issues nobody is working on. Using `Assignees` would make things easier:
- No need to spend the time writing comments to say you want to work on something
- Creating a filter for issues not assign becomes trivial
- Unassigning yourself from an issue is trivial (and I guess more likely, since having your name formally in assignees should give certain sense of responsibility)
- It becomes trivial to know if someone is assign, without having to read possibly ambiguous comments (e.g. "is it ok if I work in this?")
- I guess implementing a bot that unassignes issues after some time (a week?) of inactivity should be easy

I'm optimistic that if we start using assignees, we could possibly create 200 issues for small fixes (e.g. small changes to docstrings), label them, promote somewhere (social media, sprints...), and we should be able to provide a link where people knows what is pending (not assigned).

We'll see how this works in practice, but I think it's worth trying. I don't expect this discouraging people or making things more inefficient. But it can possibly make it easier for people who want to contribute, to find things to work on.

Reply:
sure let’s try it